src/dual.jl

```diff
@@ -314,6 +314,10 @@ end
     Integer(value(d))
 end
 
+for F in (:Float16, :Float32, :Float64)
+    @eval @inline Base.$F(d::Dual) = $F(value(d))
```

Comment:
```suggestion
    @eval @inline function Base.$F(d::Dual)
        all(iszero, partials(d)) || throw(InexactError($(nameof(F)), $F, d))
        $F(value(d))
    
```
Else we are just unexpectedly dropping the partials.


like https://github.com/JuliaDiff/ForwardDiff.jl/pull/508/files#diff-7f9840ac912342d25066b9e92a4f472a21750c5a88bf8b13232172d45bf844f4R313

Reply:
@mzgubic  said in slack that this does not work, Zygote is still broken with this
```
julia>   dx, dy = diaghessian(f34, xs, y)
ERROR: InexactError: Float64(Dual{Nothing}(0.2883004137872829,-0.36786533559880646,-0.006843046852434196,-0.0,-0.0,-0.0,-0.0))
```

Reply:
I can confirm that the following fixes Zygote

```suggestion
    @eval @inline Base.$F(d::Dual{T,V,N}) where {T,V,N} = convert(Dual{T,promote_type(V, $F),N}, d)
```

Code is based on what `AbstractFloat(::Dual)` currently does